pkg/controller/build/controller.go

```diff
@@ -274,8 +275,39 @@ func (c *Controller) syncHandler(key string) error {
 			if err != nil {
 				return err
 			}
-			if err := c.waitForOperationAsync(build, op); err != nil {
-				return err
+
+			// Check if build has timed out
```

Comment:
Will `syncHandler` be called periodically? Or only when there are updates to the build object? If it's periodic, how often is it called?

I worry that if this code is only run when a build updates (i.e., the pod's status changes), then a build that specifies one hour-long `sleep` step with a 10m timeout will not have any occasion to check for timeout until that step finishes, at which point the build has long exceeded its timeout.

Reply:
From logs it appears 30 sec is the sync interval

Reply:
Hmm, we should document that somewhere, that we effectively only enforce timeouts to a ~30s resolution. Maybe we should also `Watch` the pod, with a timeout? If we do that, do we need to worry about the controller job restarting while builds are ongoing, and losing state? (All questions for future work, not blocking this PR)